### ISSUE TITLE
Develop test functionality

### ISSUE THREAD

**Issue by leehbi** [DaxStudio/DaxStudio#368]
It's possible to write rudimentary tests in DAX Studio using a combinations of VAR for ExpectedValues, CalculatedValues

And then check values match using an IF equality check.

It would be good if there was some syntactic sugar to make this easier to  write tests and produce a table of test results.

As the model changes the user can run tests to spot any changes in measure values.

**Comment by marcosqlbi** [comment]
The issue for unit tests in DAX (or in SQL) is that the underlying database should be the same to test the DAX code integrity.
The tests might have three goals:

1. Validate the DAX calculation - the starting database should be the same

2. Validate the ETL - assuming that the source database is the same, this test validates the integrity of the pipeline ETL, so the DAX code should not use existing measures but only immutable expressions.

3. Validate the ETL + DAX - assuming that the source database is the same, this test validates the integrity of the pipeline ETL + DAX calculations (it could be redundant if 1. and 2. are implemented)

I'm not sure DAX Studio is the right tool for this, I've seen other (rudimentary) tools to implement unit tests over SSAS in the development lifecycle (search for "unit test Analysis Services").

**Comment by leehbi** [comment]
Yes - good points - you guys have done stellar work on DAX Studio.  Testing is a bit of an onion.

**Comment by dgosbell** [comment]
I'd agree with @marcosqlbi - I'm not sure if DAX Studio is the right tool for this as the problem is broader than just the tabular model. I used the framework at http://www.nbi.io/ recently and it worked reasonably well, although I found the XML syntax quite verbose
